homeassistant/helpers/normalized_name_base_registry.py

```diff
@@ -14,10 +14,14 @@ class NormalizedNameBaseRegistryEntry:
     """Normalized Name Base Registry Entry."""
 
     name: str
-    normalized_name: str
+    normalized_name: str = field(init=False)
     created_at: datetime = field(default_factory=dt_util.utcnow)
     modified_at: datetime = field(default_factory=dt_util.utcnow)
 
+    def __post_init__(self) -> None:
+        """Post init."""
+        object.__setattr__(self, "normalized_name", normalize_name(self.name))
+
```

Comment:
Not sure what the performance impact will be of having every one of these run a post init hook.  Needs to get a profile

Reply:
Seems minimal on a small system. Will need to run it on a production one.  

We don't use the base class for device registry or entity registry where it would likely matter though.  Its probably rare to have a lot of labels/areas

Reply:
Well, as far as I can see, the additional cost here is basically one more function call.
And we lose a bunch of invocations of `normalize_name` in other places, so overall the performance impact should be substantially positive.
Am I missing anything?

Reply:
I don't think so.  Its probably fine, but I need to verify it just in case I'm wrong.

Reply:
seems ok in testing